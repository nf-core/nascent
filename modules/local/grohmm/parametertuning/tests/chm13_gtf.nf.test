nextflow_process {

    name "Test Process GROHMM_PARAMETERTUNING"
    script "../main.nf"
    process "GROHMM_PARAMETERTUNING"
    tag "groHMM"

    test("Should pass with overlapping exons in GTF") {
        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id: 'Sall' ],
                    [ file("https://raw.githubusercontent.com/Kraus-Lab/groHMM/master/inst/extdata/S0mR1.bam", checkIfExists: true),
                    file("https://raw.githubusercontent.com/Kraus-Lab/groHMM/master/inst/extdata/S40mR1.bam", checkIfExists: true) ],
                    [],
                ]
                input[1] = file(
                            "https://raw.githubusercontent.com/nf-core/test-datasets/nascent/reference/broke.gtf",
                            checkIfExists: true
                            )
                input[2] = 5
                input[3] = -100
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    path(process.out.tuning[0][1]).readLines(),
                    path(process.out.bed[0][1]),
                ).match() },
            )
        }
    }
}
