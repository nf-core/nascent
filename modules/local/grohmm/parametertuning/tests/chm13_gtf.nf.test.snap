{
    "Should pass with overlapping exons in GTF": {
        "content": [
            [
                "\"LtProbB\",\"UTS\",\"merged\",\"dissociated\",\"total\",\"errorRate\",\"txSize\"",
                "-100,5,0,0,0,0,921"
            ],
            "Sall_5_-100.tuning.consensus.bed:md5,90bcbcd2a2fbd8e2c3602152d89d1cd5"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T20:16:00.717011"
    }
}